src/code/z_play.c

```diff
@@ -354,13 +353,13 @@ void Play_Init(GameState* thisx) {
         !IS_CUTSCENE_LAYER) {
         if (CHECK_QUEST_ITEM(QUEST_KOKIRI_EMERALD) && CHECK_QUEST_ITEM(QUEST_GORON_RUBY) &&
             CHECK_QUEST_ITEM(QUEST_ZORA_SAPPHIRE)) {
-            gSaveContext.sceneLayer = 1;
+            gSaveContext.sceneLayer = SCENE_LAYER_CHILD_NIGHT;
         } else {
-            gSaveContext.sceneLayer = 0;
+            gSaveContext.sceneLayer = SCENE_LAYER_CHILD_DAY;
         }
     } else if ((gEntranceTable[((void)0, gSaveContext.save.entranceIndex)].sceneId == SCENE_KOKIRI_FOREST) &&
                LINK_IS_ADULT && !IS_CUTSCENE_LAYER) {
-        gSaveContext.sceneLayer = GET_EVENTCHKINF(EVENTCHKINF_48) ? 3 : 2;
+        gSaveContext.sceneLayer = GET_EVENTCHKINF(EVENTCHKINF_48) ? SCENE_LAYER_ADULT_NIGHT : SCENE_LAYER_ADULT_DAY;
```

Comment:
```suggestion
        gSaveContext.sceneLayer = GET_EVENTCHKINF(EVENTCHKINF_48) ? 3 : 2;
```

These are not day/night layers but rather Kokiri Forest specific scene layers representing the the forest before and after the Forest Temple is completed.